src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -52,37 +52,50 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
                 statements: statements);
         }
 
-        public static IMethodSymbol CreateIEqutableEqualsMethod(
+        public static IMethodSymbol CreateIEquatableEqualsMethod(
             this SyntaxGenerator factory,
-            Compilation compilation,
+            SemanticModel semanticModel,
+            SyntaxNode typeDeclaration,
             INamedTypeSymbol containingType,
             ImmutableArray<ISymbol> symbols,
             SyntaxAnnotation statementAnnotation,
             CancellationToken cancellationToken)
         {
             var statements = CreateIEquatableEqualsMethodStatements(
-                factory, compilation, containingType, symbols, cancellationToken);
+                factory, semanticModel.Compilation, containingType, symbols, cancellationToken);
             statements = statements.SelectAsArray(s => s.WithAdditionalAnnotations(statementAnnotation));
 
-            var equatableType = compilation.GetTypeByMetadataName(typeof(IEquatable<>).FullName);
+            var equatableType = semanticModel.Compilation.GetTypeByMetadataName(typeof(IEquatable<>).FullName);
             var constructed = equatableType.Construct(containingType);
             var methodSymbol = constructed.GetMembers(EqualsName)
                                           .OfType<IMethodSymbol>()
                                           .Single(m => containingType.Equals(m.Parameters.FirstOrDefault()?.Type));
```

Comment:
note: i hate this .Single

Reply:
Separate PR, but what would you want to do instead?

Reply:
FirstOrDefault and bail sensibly on null.